tensorboard/plugins/hparams/tf_hparams_query_pane/tf-hparams-query-pane.html

```diff
@@ -320,72 +320,61 @@
+        // to downstream consumers only after all properties have been updated).
         //
-        // Note that the 'visibleSchema' field is deprecated and is included
-        // only to support the old-style schema that only contained the
-        // columns (metrics or hparams) that were displayed. It should be
-        // removed when all consumers are migrated to use the hparamColumns and
-        // metricColumns fields.
+        // Properties:
+        // -----------
+        // schema.hparamColumns[i].hparamInfo contains the HParamInfo protocol
+        // buffer representing the ith hparam.
+        // schema.metricColumns[i].metricInfo contains the MetricInfo protocol
+        // buffer representing the ith metric.
         //
-        // The 'Schema' object is initialized to an 'empty' object here and then
-        // replaced once after the getExperiment() RPC
-        // returns. After that, the schema object's will remain the
-        // same object throughout the lifetime of the page. Of course, its
-        // properties may change as a result of the user interacting with the
-        // plugin, but the "identity" of the object will remain the same.
+        // columnsVisibility[].
+        // A boolean array whose ith entry is true if
+        // the ith column is visible (selected to be displayed by the user).
+        // Columns are indexed by listing the hyperparameters first in
+        // the order they are represented by schema.hparamColumns, followed
+        // by the metrics in the order they are represented by
+        // schema.metricColumns.
         //
-        // The 'loaded' property allows consuming elements to find out whether
-        // the schema object has been computed yet or not. It is 'false' for
-        // the 'empty' object, and then set to 'true' after the getExperiment()
-        // RPC returns and the schema is computed.
-        // TODO(erez): We could have set this to a null value initially, but
-        // some consuming elements rely on visibleSchema always being not null.
+        // visibleSchema.
+        // DEPRECATED. New code should use the schema and columnVisibile
+        // properties instead.
+        // TODO(): Remove when all consumers are migrated to use the
+        // schema and columnVisibile fields.
+        // Contains arrays of HParamInfo and MetricInfo protocol buffers
+        // consisting of only the visible hyperparameters and metrics,
+        // respectively.
         /**
          * @type {{
-         *    hparamColumn: Array<{
-         *      hparamInfo: Object,
-         *      displayed: boolean
-         *    }>,
-         *    metricColumn: Array<{
-         *      metricInfo: Object,
-         *      displayed: boolean
-         *    }>,
+         *    schema: {
+         *      hparamColumns: Array<{hparamInfo: HParamInfo}>,
+         *      metricColumn: Array<{metricInfo: MetricInfo>}>,
+         *    }
+         *    columnsVisibility: Array<Boolean>,
```

Comment:
The array here can be very confusing. Instead, may I suggest, maybe in a follow up, use an object where it is keyed by name of the column (assuming that they are unique? -- I cannot imagine it not being unique as it will confuse user too :)) 

Otherwise, this user of this prop has to be aware that hparam columns appear before metric ones. 

Alternative would be, there being flat `columns` array and have a separate map/list of hparam & metrics column ids